### ISSUE TITLE
How is it possible to Get Selected Text from WebView2 (Embedded PDF Viewer) ?

### ISSUE THREAD

**Issue by IrakliLomidze** [MicrosoftEdge/WebView2Feedback#1430]
How is it possible to Get Selected Text from WebView2 (Embedded PDF Viewer)?

My WebView2 uses its own embedded PDF Viewer.
How is it possible to get a Selected text from document. ?

**Comment by champnic** [comment]
I don't believe it's currently possible to get the selected text from the embedded PDF viewer.

**Comment by IrakliLomidze** [comment]
Can I ask that feature be implemented?

**Comment by champnic** [comment]
@IrakliLomidze Done! I've opened it as a scenario on our backlog.

Unfortunately generally speaking, while we have a PDF viewer, we don't offer PDF extensibility or control through APIs, so it will probably be a while (if ever) we get to this work.

**Comment by IrakliLomidze** [comment]
Correct me if I'm wrong, PDF Viewer is a plugin based on an open-source project pdfium.
If I considering writing my own plugin is it possible to use it in WebView2?

**Comment by champnic** [comment]
I don't think so today. Instead if you have a PDF viewer you don't need to use it in WebView2. If you needed it to view documents from the web, you could listen to NavigationStarting or WebResourceRequested events for PDF files, and route it to your PDF viewer rather than letting the WebView2 display it.

**Comment by IrakliLomidze** [comment]
No, I don't have it yet. It is possible to write a .NET WinForm Wrapper of PDFium or Comiple pdfium plugin with that simple API.
Personally, I don't like both ideas because they will take a long time to implement.

I Still hope that the Edge team will expose some of the PDF API.

**Comment by ukandrewc** [comment]
@IrakliLomidze There is already a .Net wrapper for PDFium: https://github.com/xiaotie/Pdfium.Net

**Comment by vecerilek** [comment]
It would be great to have this feature.
Is there a workaround to get the selected text in the CustomItemSelected event? I need to display the associated drawing after the user selects any text in PDF.

![image](https://github.com/MicrosoftEdge/WebView2Feedback/assets/77612501/67c1c8af-4bfc-4711-88ea-e729983c66cf)

Thanks a lot.
